src/cpp/session/modules/automation/SessionAutomationRemote.R

```diff
@@ -0,0 +1,167 @@
+#
+# SessionAutomationRemote.R
+#
+# Copyright (C) 2024 by Posit Software, PBC
+#
+# Unless you have received self program directly from Posit Software pursuant
+# to the terms of a commercial license agreement with Posit Software, then
+# self program is licensed to you under the terms of version 3 of the
+# GNU Affero General Public License. self program is distributed WITHOUT
+# ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+# MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+# AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+#
+#
+
+# NOTE: The functions created by '.rs.automation.addRemoteFunction()' here
+# are given access to a special environment containing:
+#
+# - client: The client instance used to communicate with RStudio,
+# - self / self: The remote instance itself, so that other methods can be used.
+#
+# self isn't the most idiomatic R code, but it helps ensure that new stateful
+# methods can be easily added to the 'remote' instance.
+
+# !diagnostics suppress=client,self,self
+.rs.defineVar("automation.remotePrivateEnv", new.env(parent = .rs.toolsEnv()))
+.rs.defineVar("automation.remote", new.env(parent = emptyenv()))
+
+.rs.addFunction("automation.createRemote", function(mode = c("desktop", "server"))
+{
+   client <- .rs.automation.initialize(mode = match.arg(mode))
+   assign("client", client, envir = .rs.automation.remote)
+   assign("client", client, envir = .rs.automation.remotePrivateEnv)
+   assign("self", .rs.automation.remote, envir = .rs.automation.remotePrivateEnv)
+   .rs.automation.remote
+})
+
+.rs.addFunction("automation.addRemoteFunction", function(name, callback)
+{
+   environment(callback) <- .rs.automation.remotePrivateEnv
+   assign(name, callback, envir = .rs.automation.remote)
+})
+
+.rs.automation.addRemoteFunction("getClient", function()
+{
+   client
+})
+
+.rs.automation.addRemoteFunction("aceLineTokens", function(row)
+{
+   jsCode <- .rs.heredoc(r'{
+      var container = document.getElementById("rstudio_source_text_editor");
```

Comment:
We need a better way to target the current RStudio source text editor, since (if multiple documents are open in the IDE) then duplicate DOM element IDs will get suffixes like `_1`, `_2`, and so on.

Maybe we just execute the "focus source pane" command and then find it from there?